File: org.boblycat.frida.plugin.disassembler.mips/src/scala/org/boblycat/frida/plugin/disassembler/DisassemblerFactory.scala
package frida.disasm

import frida.core.FridaException

object DisassemblerFactory {

	def create(machine : String) : Disassembler = {
		machine match {
			case "mipsel" => new mips.MIPSDisasm()
			case "arm" => new arm.ARMDisasm()
			case _ => throw new FridaException("No disassembler found for machine type '" + machine + "'")
		}
	}
}

File: org.boblycat.frida.core/src/java/org/boblycat/frida/core/disassembler/DisassemblerFactory.java
package org.boblycat.frida.core.disassembler;

import java.util.HashMap;
import java.util.Map;

public class DisassemblerFactory {

	private static Map<String, Disassembler> knownDisassemblers = new HashMap<String, Disassembler>();
	
	public static void addDisassembler(String name, Disassembler newDisasm) {
		knownDisassemblers.put(name, newDisasm);
	}
	public static Disassembler create(String machine) {
		return knownDisassemblers.get(machine);
	}
}